### PR TITLE
Modify CLI release process

### PR DESCRIPTION
**Pull request by shorim**
**Description**

Changes proposed in this pull request:

- Modify the CLI release process to align with the changes done here: kyma-project/cli#992 and here: kyma-project/cli#998

**Related issue(s)**

kyma-project/cli#973

### CODE REVIEW

**Review by rakesh-garimella — DISMISSED** [review]

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ A Kyma CLI release consists of:
 * A release tag or branch holding the code
 
 ## Steps
-
+   >**NOTE:** Once `cli 2.0.0` is released, `cli 1.x.x` should not be released any more. If a new `kyma 1.24.x` is released after releasing `cli 2.0.0`, then a corresponding `cli` patch version (`2.x.x`) should be released.
 1. Make sure Kyma is released and create a release branch in the `cli` repository. The name of this branch should follow the `release-x.y` pattern, such as `release-1.4`.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:18]:
```suggestion
1. After Kyma is released, create a release branch in the `cli` repository. The name of this branch must follow the `release-x.y` pattern, such as `release-1.4`.
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 2]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
    git checkout --no-track -b {RELEASE_NAME} upstream/main
    git push upstream {RELEASE_NAME}
    ```
+2. Ensure that the `DefaultKyma1Version` variable in the `internal/config/config.go` file from the `cli` repository on the release branch contain the latest Kyma `1.24.x` version.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:27]:
```suggestion
2. Check the `internal/config/config.go` file from the `cli` repository on the release branch:
   - The `DefaultKyma1Version` variable must contain the latest Kyma `1.24.x` version.
   - Starting with Kyma 2.0.0, the `DefaultKyma2Version` variable must contain the latest Kyma `2.x.x` version.
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 3]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
 
-2. Ensure that the `KYMA_VERSION` variables in the `Makefile` and `.goreleaser.yml` files from the `cli` repository on the release branch contain the latest Kyma version.
-
-3. In `Makefile`, set `IS_RELEASE=true`
+3. Ensure that the `DefaultKyma2Version` variable in the `internal/config/config.go` file from the `cli` repository on the release branch contain the latest Kyma `2.x.x` version.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:29]:
```suggestion
```
part of step 2 now

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 4]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
+   >**NOTE:** This step applies only if at least Kyma 2.0.0 is already released.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:30]:
```suggestion
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 5]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
 
 4. Create a PR to `cli/release-x.y` that triggers the presubmit job for `cli`.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:32]:
```suggestion
3. Create a PR to `cli/release-x.y` that triggers the presubmit job for `cli`.
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 6]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
 
-5. After merging the PR, create a tag on the release branch that has the same version name as Kyma. If you define a release candidate version, a pre-release is created.  
+5. After merging the PR, create a tag on the release branch having the value of the new `cli` release version. If you define a release candidate version, a pre-release is created.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:34]:
```suggestion
5. After merging the PR, create a tag on the release branch that has the value of the new `cli` release version. If you define a release candidate version, a pre-release is created.  
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 7]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
 
-5. After merging the PR, create a tag on the release branch that has the same version name as Kyma. If you define a release candidate version, a pre-release is created.  
+5. After merging the PR, create a tag on the release branch having the value of the new `cli` release version. If you define a release candidate version, a pre-release is created.  
 
    ```bash
    git tag -a {RELEASE_VERSION} -m "Release {RELEASE_VERSION}"
    ```
 
-    where {RELEASE_VERSION} could be `1.4.0-rc1`, `1.4.0-rc2` or `1.4.0`.
+    where {RELEASE_VERSION} is the new `cli` release version and could be `2.0.0-rc1`, `2.0.0-rc2` or `2.0.0`.
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:40]:
```suggestion
    Replace {RELEASE_VERSION} with the new `cli` release version, for example, `2.0.0-rc1`, `2.0.0-rc2` or `2.0.0`.
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 8]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
 
 6. Create the new version
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:42]:
```suggestion
6. Create the new version.
```

File: docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md [hunk 9]
```diff
@@ -24,20 +24,20 @@ A Kyma CLI release consists of:
...
    1. Push the tag to trigger a postsubmit job that creates the GitHub release. Validate if the release is available under [releases](https://github.com/kyma-project/cli/releases).
```
Comment (NHingerl) [inline docs/guidelines/releases-guidelines/03-kyma-cli-release-process.md:43]:
```suggestion
   1. Push the tag to trigger a postsubmit job that creates the GitHub release. Check whether the release is available under [releases](https://github.com/kyma-project/cli/releases).
```

**Review by NHingerl — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**